### DISCUSSION TITLE
PyCharm context menu interferes with Copilot keyboard shortcut

### DISCUSSION THREAD

**Discussion by elliotwaite** [community/community#12076] [Copilot Conversations] [answered]
In PyCharm, when both the context menu and a Copilot completion are showing, pressing the keyboard shortcut for "Copilot: Apply Completions to Editor" does not work.

![screenshot](https://user-images.githubusercontent.com/1767836/155638225-0243c157-b175-416c-851f-0e2a6dd44c01.png)

Is there a way to fix this?

To see if this was an issue with the IDE rather than the plugin, I checked if the keyboard shortcuts I have for other plugins were still usable when the context menu was open, and they were. So it seems like this should be fixable from within the plugin.

**Comment by jansorg** [comment]
@elliotwaite Thanks for reporting! Can you tell what's happening instead when you press `Tab`?
By default, the shortcut is `Tab`, which is used by both PyCharm and GitHub Copilot.
Tab usually insert the completion selected in the PyCharm popup.

**Reply by elliotwaite** [reply]
When I press `Tab` it inserts the PyCharm completion. However, I set my Copilot keyboard shortcut to something other than `Tab` because I often want to accept the Copilot suggestion instead, but when I press my Copilot keyboard shortcut while the context menu is open, it seems to have no effect, the Copilot completion is not applied but remains visible, and the context menu also remains visible.

**Comment by elliotwaite** [comment, marked as answer]
I just wanted to report that this has been fixed. I'm not sure which version it was fixed in, but it's working for me in v1.1.20.1417. I can now accept the Copilot suggestion even while the context menu is open and it will automatically close the context menu and apply the Copilot suggestion. Thanks for the fix!

Also, for anyone wondering, if you want to be able to see both the Copilot suggestions (which are shown inline) and the built-in IDE's suggestions (which are shown in a context menu) at the same time so that you can choose between the two using a different keyboard shortcut for each, you have to enable this setting under preferences:

**Preferences > Languages & Frameworks > GitHub Copilot > ☑️ Show IDE completions side-by-side**
